src/main/java/org/granitepowered/granite/impl/GraniteGameRegistry.java

```diff
@@ -722,9 +763,8 @@ public ParticleEffectBuilder getParticleEffectBuilder(ParticleType particle) {
     }
 
     @Override
-    public List<GameMode> getGameModes() {
-        // TODO: GameMode API
-        throw new NotImplementedException("");
+    public ImmutableList<GameMode> getGameModes() {
+        return (ImmutableList)gamemodes.values();
```

Comment:
That will throw ClassCastException... You're supposed to do ImmutableList.copyOf(gamemodes.values(), or something like that.